docs/reference/snapshot-restore/index.asciidoc

```diff
@@ -147,30 +138,30 @@ endif::[]
 [[snapshot-index-compatibility]]
 === Index compatibility
 
-A cluster is only compatible with indices created in the previous major version
-of {es}. Any data stream or index you restore from a snapshot must be compatible
-with the current cluster's version. If you try to restore an index created in an
-incompatible version, the restore attempt will fail.
+Any index you restore from a snapshot must also be compatible with the current
+cluster's version. If you try to restore an index created in an incompatible
+version, the restore attempt will fail.
 
-A snapshot can contain indices created in a previous major version. For example,
-a snapshot of a 6.x cluster can contain an index created in 5.x. If you try to
-restore the 5.x index to a 7.x cluster, the restore attempt will fail. Keep this
-in mind if you take a snapshot before upgrading a cluster.
+include::cluster-index-compat.asciidoc[]
 
-As a workaround, you can first restore the data stream or index to another
-cluster running the latest version of {es} that's compatible with both the index
-and your current cluster. You can then use
-<<reindex-from-remote,reindex-from-remote>> to rebuild the data stream or index
-on your current cluster. Reindex from remote is only possible if the index's
-<<mapping-source-field,`_source`>> is enabled.
+A compatible snapshot can contain indices created in an incompatible version.
+For example, a snapshot of a 7.16 cluster can contain an index created in 6.8.
+If you try to restore the 6.8 index to a 8.0 cluster, the restore attempt will
+fail. Keep this in mind if you take a snapshot before upgrading a cluster.
+
+As a workaround, you can first restore the index to another cluster running the
+latest version of {es} that's compatible with both the index and your current
+cluster. You can then use <<reindex-from-remote,reindex-from-remote>> to rebuild
+the index on your current cluster. Reindex from remote is only possible if the
+index's <<mapping-source-field,`_source`>> is enabled.
```

Comment:
As a note, I updated this to remove some references to data streams. This is only applicable to indices for now.